### PR TITLE
Several fixes and changes

### LINKED ISSUE #288: Add summary at the end of the installation
```
# k8s cluster version x.yy has been successfully installed.
# Kube Server URL:  https://<SERVER_IP>:<PORT>
# Kube config: `kube get kubeconfig`
#
# Ingress controller: nginx
# Container image registry: <SERVER_IP>:<PORT>
# ....
```

### LINKED ISSUE #300: [Improvement] Show the proper registry name and port
## Improvement 

End of the execution of the kind script, the following message is displayed where `${REGISTRY_NAME}:${REGISTRY_PORT}` variables are not replaced with the proper values:
 
```
  * Log on to the container registry using the address
    docker login kind-registry:5000

✔ # Required Steps:


  * Edit the daemon.json file. If you use Docker Desktop for Mac or Docker Desktop for Windows, click the Docker icon, choose Settings and then choose Docker Engine.
    {"insecure-registries" : ["${REGISTRY_NAME}:${REGISTRY_PORT}"]}
```

### PR DESCRIPTION
**Pull request by jacobdotcosta**
* Improved script output to include include on:
  *   the container registry 
  * the `htpasswd` file
  * removal and installation steps
* Added a switch to deploy a sample application based on the `nginx` container

Closes #288 
Closes #300

### CODE REVIEW

File: kind/images.json [hunk 1]
```diff
@@ -1,4 +1,15 @@
 [
+
+  {
+    "k8s": "1.27",
+    "version": "1.27.1",
+    "sha": "kindest/node:v1.27.1@sha256:c1fea3acc2b38dd39ad2cb2ab0aa012720283e03364e314b715c395624476c74"
+  },
```
Comment (cmoulliard) [inline kind/images.json:7]: There is already an entry for 1.27, why are you adding a new one ? IF there are 2 entries for `1.27`, can the script pickup the latest of `1.27` ?

**Review by cmoulliard — CHANGES_REQUESTED** [review]

File: kind/images.json (hunk 1 above)
Comment (jacobdotcosta) [reply kind/images.json]: fixed

File: kind/kind.sh [hunk 2]
```diff
@@ -156,6 +156,7 @@ show_usage() {
     log_message "0" ""
     log_message "0" "\t--cluster-name <name>\t\t\tName of the cluster. Default: kind"
     log_message "0" "\t--delete-kind-cluster\t\t\tDeletes the Kind cluster prior to creating a new one. Default: No"
+    log_message "0" "\t--deploy-sample-application\t\tDeploys a sample application based on nginx. Default: No"
```
Comment (cmoulliard) [inline kind/kind.sh:159]: Should we do that ? Maintaining examples up to date with what we package/install is not so easy !

**Review by cmoulliard — COMMENTED** [review]
I dont see part of the http-server.yaml the resource to create the ingress route (if we keep the sample of course part of this bash script)

File: kind/kind.sh (hunk 2 above)
Comment (jacobdotcosta) [reply kind/kind.sh:159]: I see the problem. I'll remove the sample then.

Comment (jacobdotcosta) [reply kind/kind.sh]: Removed the sample application deployment from the kind installation process.

**Comment by jacobdotcosta** [comment]
Closes #300

**Comment by jacobdotcosta** [comment]
Added an option to set the registry name `--registry-name <name>			Name of the registry. Default: <cluster_name>-registry`.

@cmoulliard would this fix #301 ?

**Review by cmoulliard — APPROVED** [review]

**Pull request merged**